### DISCUSSION TITLE
ARCCE: anyone still using the old interfaces?

### DISCUSSION THREAD

**Discussion by aldbr** [DIRACGrid/DIRAC#7551] [General]
Is anyone still using the [`ARCComputingElement`](https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Resources/Computing/ARCComputingElement.py) or [`ARC6ComputingElement`](https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Resources/Computing/ARC6ComputingElement.py)?

If not, we could actually drop them from `v9.0`.

**Comment by IgorPelevanyuk** [comment]
Hello @aldbr ,

Whe have CEType=ARC. Could you please tell how chould I change this line to work with curent interface for ARC resources?

**Reply by aldbr** [reply]
Sure:

- step1: from a DIRAC client, execute the following command to get the port used by the CE to provide the AREX interface:

```bash
$ arcinfo -c <ce name>
Computing service: arc (production)
  Information endpoint: ldap://<ce name>:2135/Mds-Vo-Name=local,o=grid (org.nordugrid.ldapng)
  Information endpoint: ldap://<ce name>:2135/o=glue (org.nordugrid.ldapglue2)
  Information endpoint: https://<ce name>:443/arex (org.nordugrid.arcrest) <------------ get the port used here
  Information endpoint: https://<ce name>:443/arex (org.ogf.glue.emies.resourceinfo)
  Submission endpoint: https://<ce name>:443/arex (status: ok, interface: org.nordugrid.arcrest)
  Submission endpoint: https://<ce name>:443/arex (status: ok, interface: org.ogf.glue.emies.activitycreation)
  Submission endpoint: gsiftp://<ce name>:2811/jobs (status: ok, interface: org.nordugrid.gridftpjob)
```

- step2: simply replace `CEType=ARC` with `CEType=AREX` in the configuration
- step3: if the port is not `443` (the default one), then you also need to add the following option to your configuration:

```bash
Port = <AREX port>
```

Please let me know if you need any help.
Thanks

**Reply by aldbr** [reply]
If you encounter such logs from the Site Director execution:

```
Request timed out, consider increasing the Timeout value...
```

Follow the suggestion by adding a `Timeout` option in the CE configuration with a value higher than `5`, for instance:

```
Timeout = 60
```

**Reply by IgorPelevanyuk** [reply]
Thank you! Our ARC resources are working with AREX now.

**Comment by marianne013** [comment]
Just to say that GridPP is all AREX.

**Comment by iueda** [comment]
As was reported at the last DUW (https://indico.cern.ch/event/1341205/contributions/5972938/)
we have sites with ARC-CE which do not seem to be accessible with AREX, and need to talk to them

**Reply by aldbr** [reply]
Only the sites mentioned in the slides or more of them?
Because:
- it looks like `pikolit.ijs.si` and `hpc.arnes.si` provide an AREX interface:

```
> arcinfo -c pikolit.ijs.si
Computing service: SiGNET (production)
  ...
  Information endpoint: https://pikolit.ijs.si:443/arex (org.nordugrid.arcrest)
  Submission endpoint: https://pikolit.ijs.si:443/arex (status: ok, interface: org.nordugrid.arcrest)
> arcinfo -c hpc.arnes.si
Computing service: ARNES (production)
  ...
  Information endpoint: https://hpc.arnes.si:443/arex (org.nordugrid.arcrest)
  Submission endpoint: https://hpc.arnes.si:443/arex (status: ok, interface: org.nordugrid.arcrest)
```

- we are using the AREX interface to interact with `arcce03.esc.qmul.ac.uk` (we switched in november 2023)

**Reply by iueda** [reply]
Indeed, I got reported that SiGNET is now ok.

Then, we have the following with failing AREX access
arcce02.esc.qmul.ac.uk
arcce03.esc.qmul.ac.uk

**Reply by aldbr** [reply]
Can you use these instances? IIUC, QMUL is in [downtime](https://goc.egi.eu/portal/index.php?Page_Type=Downtime&id=35178) until September, we cannot access the CEs for now.

**Reply by spardi** [reply]
Before the down we tested the CE as follow
arcce02.esc.qmul.ac.uk  AREX
arcce03.esc.qmul.ac.uk  ARC
I'll check the situation after the donwtime.

**Comment by marianne013** [comment]
I can confirm QMUL is in downtime due to a data center refurbishment, but we've been using their AREX interfaces for a while now, and there is no reason why they wouldn't work after they come out of downtime.
